tree/ntuple/v7/src/RNTupleModel.cxx

```diff
@@ -318,6 +322,43 @@ void ROOT::Experimental::RNTupleModel::AddField(std::unique_ptr<RFieldBase> fiel
    fFieldZero->Attach(std::move(field));
 }
 
+void ROOT::Experimental::RNTupleModel::AddSubfield(std::string_view qualifiedFieldName, REntry &entry,
+                                                   bool initializeValue) const
+{
+   auto field = FindField(qualifiedFieldName);
+   if (initializeValue)
+      entry.AddValue(field->CreateValue());
+   else
+      entry.AddValue(field->BindValue(nullptr));
+}
+
+void ROOT::Experimental::RNTupleModel::RegisterSubfield(std::string_view qualifiedFieldName)
+{
+   if (qualifiedFieldName.empty())
+      throw RException(R__FAIL("no field name provided"));
+
+   if (fRegisteredSubfields.find(std::string(qualifiedFieldName)) != fRegisteredSubfields.end())
```

Comment:
I think we should also check that we register a subfield, and not a top-level field.